cmake/alpakaCommon.cmake

```diff
@@ -178,6 +178,15 @@ else()
             target_link_libraries(alpaka INTERFACE ${RT_LIBRARY})
         endif()
     endif()
+
+    # Add debug optimization levels. CMake doesn't do this by default.
+    # Note that -Og is the recommended gcc optimization level for debug mode but is equivalent to -O1 for clang (and its derivates).
+    alpaka_set_compiler_options(HOST_DEVICE target alpaka "$<$<AND:$<CONFIG:Debug>,$<CXX_COMPILER_ID:GNU>>:-Og>"
+                                                          "$<$<AND:$<CONFIG:Debug>,$<CXX_COMPILER_ID:Clang,AppleClang,IntelLLVM>>:-O0>"
+                                                          "$<$<AND:$<CONFIG:Debug>,$<CXX_COMPILER_ID:MSVC>>:/Od>")
```

Comment:
See the comment above. `-Og` is equivalent to `-O1` for LLVM compilers so it might affect debug information. The GNU compiler recommends `-Og` because the enabled optimizations don't hurt the debug experience.